sync.go

```diff
@@ -113,7 +127,9 @@ func (lw *LibWallet) SpvSync(peerAddresses string) error {
 
 	// syncer.Run uses a wait group to block the thread until sync completes or an error occurs
 	go func() {
+		lw.syncing = true
 		err := syncer.Run(ctx)
+		lw.syncing = false
```

Comment:
I think I'd find it easier to deduce from the following snippet that `lw.syncing` is turned off when the go routine is exited.
```
lw.syncing = true
defer func() {
	lw.syncing = false
}()

err := syncer.Run(ctx)
```

Also need to set `lw.syncing = false` in following callbacks:
- sync canceled
- sync error
- sync completed